### DISCUSSION TITLE
Annis via Apache HTTP Server

### DISCUSSION THREAD

**Discussion by gcelano** [korpling/ANNIS#856] [General]
If Annis is made available through Apache HTTP Server at `myurl.de/annis/`, it cannot work properly (see attached screenshot; but it works if it is served at myurl.de). This has to do with the `/annis/` addition, which interferes somewhow with the URLs generated by Annis. I tried a number of `RewriteRule` in Apache, but nothing works: any suggestion?

Relevant directive:
```
<Location /annis/>
   ProxyPass http://localhost:5712/
   ProxyPassReverse http://localhost:5712/
</Location>
```
**Desktop (please complete the following information):**
 - Debian 12
 - openjdk 17.0.11
 - ANNIS Version 4.11
 

![Screenshot from 2024-05-27 12-54-16](https://github.com/korpling/ANNIS/assets/5678472/9b7dfca9-601a-4b72-83fe-6911ae9456d3)

**Comment by thomaskrause** [comment]
If you use a sub-path to access ANNIS, rewriting the URL in Apache alone does not help because the generated HTML code also assumes as certain path.

You can overwrite which context path is used by adding the following line to the `application.properties`  file

```
server.servlet.context-path=/annis
```

The file can be located at several places, and you could set this property also as a command line argument (<http://korpling.github.io/ANNIS/4/user-guide/configuration/index.html>).
